mono/profiler/proflog.c

```diff
@@ -1144,15 +1142,15 @@ remove_thread (MonoProfiler *prof, MonoProfilerThread *thread, gboolean from_cal
 
 			InterlockedIncrement (&thread_ends_ctr);
 
-			buffer = ensure_logbuf_inner (buffer,
+			thread->buffer = ensure_logbuf_inner (thread->buffer,
```

Comment:
This plus a few other crash/deadlock fixes in this PR are definitely worth backporting.